### PR TITLE
Add feature to push manifest

### LINKED ISSUE #53: Add `manifest` input
Add feature to push manifest.

### PR DESCRIPTION
**Pull request by divyansh42**
Signed-off-by: divyansh42 <diagrawa@redhat.com>

### Description
Manifest will be detected automatically by this action, no separate input is required for this feature.

### Related Issue(s)
Close #53 

### Checklist

- [x] This PR includes a documentation change
- [ ] This PR does not need a documentation change
---
- [x] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is not user-facing
- [ ] This change is a patch change
- [x] This change is a minor change
- [ ] This change is a major (breaking) change

### CODE REVIEW

**Comment by divyansh42** [comment]
https://github.com/divyansh42/push-to-registry/actions/runs/1445395283 this workflow run describes manifest build and push

File: src/index.ts [hunk 1]
```diff
@@ -107,87 +107,92 @@ async function run(): Promise<void> {
     }
 
     const registryPathList: string[] = [];
+    // here
+    // check if provided image is manifest or not
+    const isManifest = await checkIfManifestsExists();
```
Comment (tetchel) [inline src/index.ts:112]: log any images that are manifests

File: src/index.ts [hunk 2]
```diff
@@ -392,6 +402,38 @@ async function removeDockerPodmanImageStroage(): Promise<void> {
     }
 }
 
+async function checkIfManifestsExists(): Promise<boolean> {
```
Comment (tetchel) [inline src/index.ts:405]: If at least one manifest is found, log somewhere the list of images that are manifests

File: src/index.ts [hunk 3]
```diff
@@ -392,6 +402,38 @@ async function removeDockerPodmanImageStroage(): Promise<void> {
...
+        );
+        if (commandResult.exitCode === 0) {
+            foundManifests.push(manifest);
+        }
+        else {
+            missingManifests.push(manifest);
+        }
+    }
+
+    if (foundManifests.length > 0 && missingManifests.length > 0) {
```
Comment (tetchel) [inline src/index.ts:424]: So it is an error condition if some of the images are manifests, but not all of them ? is that the right behaviour? what if some are regular images?

File: src/index.ts [hunk 4]
```diff
@@ -392,6 +402,38 @@ async function removeDockerPodmanImageStroage(): Promise<void> {
...
+        throw new Error(`Manifest${missingManifests.length !== 1 ? "s" : ""} "${missingManifests.join(", ")}" `
+            + `not found in the Podman image storage.`);
+    }
+
+    if (missingManifests.length === sourceImages.length) {
```
Comment (tetchel) [inline src/index.ts:429]: I find this kind of confusing.

I would do `return foundManifests.length === sourceImages.length`.

**Review by tetchel — CHANGES_REQUESTED** [review]

**Comment by tetchel** [comment]
Also needs a readme update

File: README.md [hunk 5]
```diff
@@ -83,6 +83,13 @@ For example:
 
 `registry-path`: The first element of `registry-paths`, as a string.
 
+## Pushing Manifest
+
+This action supports manifest, this means you can push the manifest with this action.
```
Comment (tetchel) [inline README.md:88]: Please link to some buildah documentation explaining what manifests are

File: README.md [hunk 6]
```diff
@@ -12,7 +12,7 @@
 [![license badge](https://img.shields.io/github/license/redhat-actions/push-to-registry)](./LICENSE)
 [![size badge](https://img.shields.io/github/size/redhat-actions/push-to-registry/dist/index.js)](./dist)
 
-Push-to-registry is a GitHub Action for pushing a container image to an image registry, such as Dockerhub, quay&#46;io, the GitHub Container Registry, or an OpenShift integrated registry.
+Push-to-registry is a GitHub Action for pushing a container image or a manifest to an image registry, such as Dockerhub, quay&#46;io, the GitHub Container Registry, or an OpenShift integrated registry.
```
Comment (tetchel) [inline README.md:15]: Needs explanation of what a manifest is (link to buildah doc)

**Review by tetchel — APPROVED** [review]

**Pull request merged**